llvm/PKGBUILD

```
@@ -0,0 +1,289 @@
+            'cd3b441073e007b2e2900ea778be96076286178b7c1c6db0f2ef43fe1836906e'
+            'a41246e6d0c7ee74dda2dc6bb99966b0523d9b1718d4ab330ee3d9e05e57fd41')
+validpgpkeys=('B6C8F98282B944E3B0D5C2530FC3042E345AD05D'  # Hans Wennborg, Google.
+              '474E22316ABF4785A88C6E8EA2C794A986419D8A'  # Tom Stellard
+              'D574BD5D1D0E98895E3BF90044F2485E45D59042') # Tobias Hieta
+noextract=(clang-${pkgver}.src.tar.xz
+           runtimes-${pkgver}.src.tar.xz)
+
+apply_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying ${_patch}"
+    patch -Nbp1 -i "${srcdir}/${_patch}"
+  done
+}
+
+revert_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Reverting ${_patch}"
+    patch -p1 -i "${srcdir}/${_patch}"
+  done
+}
+
+apply_git_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying ${_patch}"
+    patch -Nbp2 -i "${srcdir}/${_patch}"
+  done
+}
+
+prepare() {
+  cd "${srcdir}"
+  for pkg in clang runtimes; do
+    plain "Extracting ${pkg}-${pkgver}.src.tar.xz due to symlink(s) without pre-existing target(s)"
+    tar -xJf "${srcdir}"/${pkg}-${pkgver}.src.tar.xz -C "${srcdir}" || true
+  done
+
+  # Rename Directories
+  # We need libunwind for LLD: https://github.com/llvm/llvm-project/issues/48572
+  for pkg in llvm clang compiler-rt lld cmake libunwind runtimes; do
+    mv ${pkg}-$pkgver.src ${pkg}
+  done
+
+  # Patch llvm
+  cd "${srcdir}"/llvm
+  apply_git_patch_with_msg 0001-LLVM-Cygwin-Fix-symbol-visibility-definition.patch
+  apply_git_patch_with_msg 0002-undef-i386.patch
+
+  # Patch clang
+  cd "${srcdir}"/clang
+  apply_git_patch_with_msg 0101-hack-dynamically-resolve-G-include-dir.patch
+  apply_git_patch_with_msg 0102-Clang-Cygwin-Enable-few-conditions-that-are-shared-w.patch
+  apply_git_patch_with_msg 0103-Clang-Cygwin-Enable-TLS.patch
+  apply_git_patch_with_msg 0104-Clang-Cygwin-Fix-symbol-visibility-definition.patch
+  apply_git_patch_with_msg 0105-hack-cygwin-use-proper-EH-model.patch
+  apply_git_patch_with_msg 0106-disable-shared-libs-on-Cygwin-by-default.patch
+  apply_git_patch_with_msg 0107-Clang-Cygwin-Remove-erroneous-define.patch
+
+  # Patch lld
+  cd "${srcdir}"/lld
+  apply_git_patch_with_msg 0201-LLD-MinGW-Implement-dll-search-prefix-option.patch
+  apply_git_patch_with_msg 0202-hack-add-system-search-paths.patch
+  apply_git_patch_with_msg 0203-hack-add-data-bss-start-end.patch
+}
+
+build() {
+  local -a platform_config
+  local -a common_cmake_args
+
+  if check_option "debug" "y"; then
+    common_cmake_args+=(-DCMAKE_BUILD_TYPE=Debug
+                        -DLLVM_ENABLE_ASSERTIONS=ON)
+    VERBOSE="VERBOSE=1"
+  else
+    common_cmake_args+=(-DCMAKE_BUILD_TYPE=Release
+                        -DLLVM_ENABLE_ASSERTIONS=OFF
+                        -DLLVM_ENABLE_DUMP=ON)
+  fi
+  common_cmake_args+=(-Wno-dev
+    -DCMAKE_INSTALL_PREFIX=/usr
+    -DCOMPILER_RT_BUILD_ORC=OFF
+    -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON
+    -DLLVM_DEFAULT_TARGET_TRIPLE=${CARCH}-pc-cygwin
```

I wonder if this shouldn't be
```suggestion
    -DLLVM_DEFAULT_TARGET_TRIPLE=${CARCH}-pc-windows-cygnus
```
as the normalized triple

---

oops, should have done the same to the host triple too, I missed that